Stage 1 Frontend and API Submission

kedycatsudo opened the pull request:
Submitting project for bootcamp review as instructed. All features implemented as required.

kedycatsudo committed:
Bootcamp submission note

kedycatsudo committed:
register navigation at login.jsx fixed.

gennady-bars reviewed (commented):
(**Please, expand the general comment ↓**)

Dear Doruk, 

Unfortunately, I am **unable to complete the review** at this time because:

* https://snipboard.io/iJexn0.jpg In **Files changed**, there must be the code that should be reviewed. It means that **all changes** from the last sprint should be there. But now there is only 1 file there. I'm not able to leave comments in your code. 



If you need more assistance, I recommend booking a 1:1 session with a tutor who can walk you through the tricky parts and clarify your questions in real time. You can book a session here: [https://calendly.com/tripleten-se-experts ](https://calendly.com/tripleten-se-experts%20)

If your code is **correct locally** on your computer, then check if you have **pushed** all the changes to `github`. I can see those changes only if they are in your branch

kedycatsudo committed:
Resolve merge conflict in login page

kedycatsudo commented:
Hello @gennady-bars. I didn`t get exactly what you mean by:
In Files changed, there must be the code that should be reviewed. It means that all changes from the last sprint should be there. But now there is only 1 file there. I'm not able to leave comments in your code.

That is my custom final project and i follow those steps from final project sprint:
5. Project submission

Submit the link to the pull request opened from the stage-1-frontend-and-api branch to main.

Make sure you attach the link leading specifically to the pull request, not to the repository. Once the work is approved by an instructor, click the "Merge" button to apply the changes from the stage-1-frontend-and-api branch to main. 

I realized that there was problems that i need to fix in my code, i just fixed it on main branch because the project running live on github pages from main branch. But then i merge it to this branch:  Stage 1 Frontend and API Submission.
And now the branch works without error and as expected. And now i am submitting the same pull request again because i fixed the problems that i found in this branch : Stage 1 Frontend and API Submission. 
Correct me wherever i am wrong please. Thank you for your time.

kedycatsudo commented:
So you are the first one that will review this project, i fixed stuff that i realized on my project after i publish it to github pages. Hope that is make sense.

kedycatsudo committed:
Read the README.md

n817 commented:
Hello! I'm glad to say that you've done a really good job! Your project has been accepted.
I wish you success in the next stage and in your future projects!✨

Number of points: 93

# Summary:
### Performance Criteria

#### Markup and JSX
- [x] <!--7.4--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--7.26--> Navigation between pages and links to external resources are working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.
- [x] <!--7.26--> Elements specified in the UI toolkit are used correctly (e.g., buttons, forms, text elements, etc).
- [ ] <!--3.63--> Classes are named according to BEM specifications.

* According to the BEM rules, an element is always part of a block, and you shouldn't use it separately from the block.
E.g. the `explore__food-header` element should be inside the `explore` block, the `header__container` element should be inside the `header` block etc.:
https://en.bem.info/methodology/quick-start/#membership
Please fix all similar cases throughout the project.

- [x] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--3.63--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--3.63--> Infrastructural project files are created using CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [x] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.

- [ ] <!--3.63--> There are no linting errors or warnings.

* There are some linting errors in the project. You can check them by running `npm run lint`.

#### React
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [x] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`.
- [x] <!--3.63--> No third-party JavaScript libraries are used, unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [x] <!--1.25--> The icons are exported in SVG format.
- [x] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [x] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with `addEventListener` are removed when the component is unmounted.
- [x] <!--1.25--> API error handling:
  - [x] The user receives a message in the case of an error.
- [x] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
- [x] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed using the cross button, overlay, or ESC key.

### Recommendations

#### Markup and JSX
- [x] <!--0.83--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [x] <!--0.83--> Raster and vector images have been optimized.

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [x] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks.
- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and clearly structured. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, when a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, and the indentation hierarchy is respected.
